build-capa.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+set -e
+function subst_env_vars {
+    local ORIG_FILE="$1"
+    local T_FILE="$2"
+    cp "$ORIG_FILE" "$T_FILE"
+    declare -a subst=(
+      "AWS_B64ENCODED_CREDENTIALS=__.Values.aws.encodedCredentials__"
+      "AWS_CONTROLLER_IAM_ROLE/="
+      "AWS_CONTROLLER_IAM_ROLE=__.Values.aws.iamRole__"
+      "K8S_CP_LABEL:=node-role.kubernetes.io/control-plane"
+      "CAPA_DIAGNOSTICS_ADDRESS=:8443"
+      "CAPA_INSECURE_DIAGNOSTICS=false"
+      "CAPA_LOGLEVEL=0"
+      "EXP_MACHINE_POOL=true"
+      "CAPA_EKS=true"
+      "CAPA_EKS_IAM=false"
+      "CAPA_EKS_ADD_ROLES=false"
+      "EXP_EKS_FARGATE=false"
+      "EVENT_BRIDGE_INSTANCE_STATE=false"
+      "AUTO_CONTROLLER_IDENTITY_CREATOR=false"
+      "EXP_BOOTSTRAP_FORMAT_IGNITION=false"
+      "EXP_EXTERNAL_RESOURCE_GC=false"
+      "EXP_ALTERNATIVE_GC_STRATEGY=false"
+      "TAG_UNMANAGED_NETWORK_RESOURCES=false"
+      "EXP_ROSA=true"
+    )
+    for SUB in "${subst[@]}" ; do
+       local WHAT="${SUB%=*}"
+       local WITH="${SUB##*=}"
+       sed -i "s;\${$WHAT[^}]*};$WITH;g" "$T_FILE" 
+    done
+}
+function subst_placeholders {
+    local T_FILE="$1"
+    #sed -i -e 's/\({{\|}}\)/{{ "\1" }}/g' "$T_FILE"
+    sed -i -e 's;__\(.Values.[^_]\+\)__;{{ \1 }};g' "$T_FILE"
+    TMP_F=$(mktemp)
+    while true ; do
+        F_LINE=$(grep '[[:space:]]*\(- \|\)'"'"'*z.._.READ_FILE._:' "$T_FILE"|head -1)
+        [ -z "$F_LINE" ] && break
+        INDENT="${F_LINE%%z??_*}"
+        INDENT="${INDENT%%\'}"
+        INDENT="${INDENT%%\ \ -\ }"
+        R_FILE="${F_LINE#*z??_.READ_FILE._:\ }"
+        R_FILE="${R_FILE%\'}"
+        R_FILE="src/$PRJ/placeholders/add/$R_FILE"
+        sed -e "s;^;$INDENT;" "$R_FILE" > $TMP_F
+        sed -i -e "/^${F_LINE}$/r $TMP_F" -e "//d" "$T_FILE"
+    done
+    rm -f "$TMP_F"
+}
+function remove_cert_manager {
+    local T_FILE="$1"
+    if [ "${USE_OC_CERT=yes}" == "yes" ] ; then
+        $YQ -i 'select(.metadata.annotations."cert-manager.io/inject-ca-from"=="capa-system/capa-serving-cert").metadata.annotations."service.beta.openshift.io/inject-cabundle"="true"' "$T_FILE"
+        $YQ -i 'del(.metadata.annotations."cert-manager.io/inject-ca-from")' "$T_FILE"
+        $YQ -i 'del(.spec.conversion.webhook.clientConfig.caBundle)' "$T_FILE"
```

Comment:
please use kustomize to do overlay OR remove the yaml structure. Better to minimize the shell work

Reply:
I understand.
I have tried this, but I cannot test the annotation=value match because I was unable to match a value that includes a slash character :(